cmd/fieldversiontag/fieldversiontag.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/component-base/featuregate"
+	kfeatures "k8s.io/kubernetes/pkg/features"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+)
+
+var (
+	typeSrc              = flag.StringP("type-src", "s", "", "From where we are going to read the types")
+	featureGateRegex     = regexp.MustCompile(`featureGate=([A-Za-z]*)`)
```

Comment:
May I suggest using a stricter regex matching to be less ambiguous and avoid conflicts with any (future) comments that could include the keywords "featureGate", "minVersion" or "prerlease".

```suggestion
	featureGateRegex     = regexp.MustCompile(`\+k8s:openapi-gen:[^+]*featureGate=([A-Za-z]*)`)
	minVersionRegex      = regex.MustCompile(`\+k8s:openapi-gen:[^+]*minVersion=([A-Za-z0-9.]*)`)
	prereleaseRegex       = regexp.MustCompile(`\+k8s:openapi-gen:[^+]*prerelease=([A-Za-z]*)`) 
```